oci/table_oci_dns_tsig_key.go

```diff
@@ -0,0 +1,252 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v36/common"
+	"github.com/oracle/oci-go-sdk/v36/dns"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableDnsTsigKey(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_dns_tsig_key",
+		Description: "OCI DNS TSIG Key",
+		List: &plugin.ListConfig{
+			Hydrate: listDnsTsigKeys,
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getDnsTsigKey,
+		},
+		GetMatrixItem: BuildCompartmentList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "A globally unique domain name identifying the key for a given pair of hosts.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the resource.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the resource was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+
+			// other columns
+			{
+				Name:        "algorithm",
+				Description: "TSIG key algorithms are encoded as domain names, but most consist of only one non-empty label, which is not required to be explicitly absolute.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "secret",
+				Description: "A base64 string encoding the binary shared secret.",
```

Comment:
What type of info is in this secret? Am I able to base 64 decode this info and get sensitive information?

Reply:
The secret is the base 64 encoding of the binary TSIG key. we can decode it and get information. To access dns security access is required . https://docs.oracle.com/en-us/iaas/Content/DNS/Concepts/dnszonemanagement.htm